codecs/src/aper/encode/encode_internal.rs

```diff
@@ -20,6 +20,13 @@ pub(super) fn encode_semi_constrained_whole_number(
     lb: i128,
     value: i128,
 ) -> Result<(), AperCodecError> {
+    if value < lb {
+        return Err(AperCodecError::new(format!(
+            "Cannot encode integer {} - less than lower bound {}",
+            value, lb,
+        )));
+    }
+
```

Comment:
This is right +1 I missed out on my original description, but later remembered, this will be required.